src/core/main.c

```diff
@@ -1977,13 +1975,13 @@ static int invoke_main_loop(
         assert(ret_error_message);
 
         for (;;) {
-                r = manager_loop(m);
-                if (r < 0) {
+                ManagerObjective obj = manager_loop(m);
```

Comment:
manager_loop() returns `int`, not `ManagerObjective`?

`obj` is a weird name. I am pretty sure most people would assume it refers to an "object" of some form, not an "objective".

Maybe name it "objective"? or "ov"?